servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcProviders.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © 2022 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.grpc.api;
+
+import io.servicetalk.http.api.HttpProviders;
+import io.servicetalk.http.api.HttpProviders.HttpServerBuilderProvider;
+import io.servicetalk.http.api.HttpProviders.SingleAddressHttpClientBuilderProvider;
+
+import java.net.SocketAddress;
+import java.util.ServiceLoader;
+
+/**
+ * A holder for all gRPC-specific providers that can be registered using {@link ServiceLoader}.
+ *
+ * @see HttpProviders
+ */
+public final class GrpcProviders {
```

Comment:
When I think of the name `Provider` it is akin to me to `Supplier`, but extended from no-args to act as an abstracted virtual constructor. These interfaces feel more like `Customizers` to use Spring terminology. I wonder if others might be confused by the name.

Reply:
"provider" is a term defined by `ServiceLoader` javadoc. Since all "providers" are tight to `ServiceLoader`, using its terminology vs Spring terminology is more natural. Armeria also uses the name "provider".